lib/resolve-value.js

```diff
@@ -41,12 +39,19 @@ var resolveValue = function(decl, map, /*optional*/ignorePseudoScope, /*internal
 
 	// Resolve any var(...) substitutons
 	var isResultantValueUndefined = false;
-	resultantValue = resultantValue.replace(new RegExp(RE_VAR_FUNC.source, 'g'), function(match, variableName, fallback) {
-		// Loop through the list of declarations for that value and find the one that best matches
-		// By best match, we mean, the variable actually applies. Criteria:
-		//		- is under the same scope
-		//		- The latest defined `!important` if any
-		var matchingVarDeclMapItem;
+	
+	// var() = var( <custom-property-name> [, <any-value> ]? )
+	// matches `name[, fallback]`, captures "name" and "fallback"
+	// See: http://dev.w3.org/csswg/css-variables/#funcdef-var
+	var match;
+	while (match = balanced('var(', ')', resultantValue)) {
+		var matchingVarDeclMapItem = undefined;
+
+		// split comma to find variable name and fallback value
+		match.body = match.body.split(',');
+		// get variable name and fallback, filtering empty items
+		var [ variableName, fallback ] = [ match.body.shift(), match.body.join(',') ].map(item => item.trim()).filter(item => !!item);
```

Comment:
Let's avoid the destructuring syntax just to keep with the maybe Node.js compatibility some people like to have.

This also has a bit of necessary array magic. We should opt for something easier to understand.

Is this the same? Seems like we don't need to trim things  and can leave peoples formatting alone.

```js
const variableName = variableFallbackSplitPieces[0].trim();
const fallback = variableFallbackSplitPieces.slice(1).join(',');
```

Reply:
done, changed it for better reading